stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/RepositoryFacade.kt

```diff
@@ -73,13 +75,35 @@ internal class RepositoryFacade constructor(
     }
 
     override suspend fun insertChannel(channel: Channel) {
-        channelsRepository.insertChannel(channel)
         insertUsers(channel.let(Channel::users))
+        channelsRepository.insertChannel(channel)
     }
 
     override suspend fun insertChannels(channels: Collection<Channel>) {
-        channelsRepository.insertChannels(channels)
         insertUsers(channels.flatMap(Channel::users))
+        channelsRepository.insertChannels(channels)
+    }
+
+    override suspend fun insertMessage(message: Message, cache: Boolean) {
+        insertUsers(message.users())
```

Comment:
it it odd that insertUsers is called inside insertMessage, this is causing a side effect that I wouldn't normally expect.

Reply:
It is because `Message` has an entity `user` inside of it, but our DB doesn't store nested object, so we need to be sure that our nested object are stored on the proper table